examples/auto-log-subscription/main.tf

```diff
@@ -0,0 +1,36 @@
+resource "random_pet" "this" {}
+
+# This is the bucket where data will be written to.
+resource "aws_s3_bucket" "this" {
+  bucket_prefix = "${random_pet.this.id}-"
+  force_destroy = true
+}
+
+# We will include this log group in our subscription
+resource "aws_cloudwatch_log_group" "included" {
+  name_prefix = "${random_pet.this.id}-"
```

Comment:
Better to not enforce random naming here. People can refer these examples as modules directly for quick bootstrapping scenarios and being able to specify a name_prefix explicitly would be a plus. The default can still be random.

Reply:
This example modules don't have any input variables, so they shouldn't be referenced as modules. Are you suggesting we use locals to decouple the reference to a resource?

Reply:
Updated the examples with `local.name` and `local.name_prefix` respectively.

Reply:
My main point was that nothing stops one from doing:
```
module "example_collection" {
  source = "git::https://github.com/observeinc/terraform-aws-collection//examples/auto-log-subscription"
}
```
It's generally a pattern in terraform repos to directly refer examples e.g.: See the code snippet for https://registry.terraform.io/modules/terraform-aws-modules/vpc/aws/latest/examples/complete